graphql_api/types/flake_aggregates/flake_aggregates.py

```diff
@@ -1,37 +1,88 @@
-from typing import TypedDict
+from dataclasses import dataclass
 
+import polars as pl
 from ariadne import ObjectType
 from graphql import GraphQLResolveInfo
+from shared.django_apps.core.models import Repository
 
-flake_aggregates_bindable = ObjectType("FlakeAggregates")
+from utils.test_results import get_results
 
 
-class FlakeAggregate(TypedDict):
+@dataclass
+class FlakeAggregates:
     flake_count: int
-    flake_count_percent_change: float | None
     flake_rate: float
-    flake_rate_percent_change: float | None
+    flake_count_percent_change: float | None = None
+    flake_rate_percent_change: float | None = None
+
+
+def calculate_flake_aggregates(table: pl.DataFrame) -> pl.DataFrame:
```

Comment:
is it possible to know what columns exist on the table at this point? Right now it's a little "magic"